doc/schema/libdoc.02.xsd

```diff
@@ -0,0 +1,93 @@
+    </xs:sequence>
+    <xs:attribute name="name" type="xs:string" use="required" />
+    <xs:attribute name="type" type="LibraryType" use="required" />
+    <xs:attribute name="format" type="DocumentationFormat" use="required" />
+    <xs:attribute name="scope" type="LibraryScope" use="required" />
+    <xs:attribute name="namedargs" type="xs:boolean" use="required" />
+    <xs:attribute name="generated" type="GenerationDate" use="required" />
+    <xs:attribute name="specversion" type="SpecVersion" use="required" />
+    <xs:attribute name="source" type="xs:string" use="optional" />
+    <xs:attribute name="lineno" type="xs:positiveInteger" use="optional" />
+  </xs:complexType>
+  <xs:complexType name="Init">
+    <xs:sequence>
+      <xs:element name="arguments" type="Arguments" />
+      <xs:element name="doc" type="xs:string" />
+      <xs:element name="tags" type="Tags" />
+    </xs:sequence>
+    <xs:attribute name="deprecated" type="xs:boolean" use="optional" />
+    <xs:attribute name="source" type="xs:string" use="optional" />
+    <xs:attribute name="lineno" type="xs:positiveInteger" use="optional" />
+  </xs:complexType>
+  <xs:complexType name="Keyword">
+    <xs:sequence>
+      <xs:element name="arguments" type="Arguments" />
+      <xs:element name="doc" type="xs:string" />
+      <xs:element name="tags" type="Tags" />
+    </xs:sequence>
+    <xs:attribute name="name" type="xs:string" use="required"/>
+    <xs:attribute name="deprecated" type="xs:boolean" use="optional" />
+    <xs:attribute name="source" type="xs:string" use="optional" />
+    <xs:attribute name="lineno" type="xs:positiveInteger" use="optional" />
+  </xs:complexType>
+  <xs:complexType name="Arguments">
+    <xs:sequence>
+      <xs:element name="arg" type="xs:string" minOccurs="0" maxOccurs="unbounded" />
+    </xs:sequence>
+  </xs:complexType>
+  <xs:complexType name="Tags">
+    <xs:sequence>
+      <xs:element name="tag" type="xs:string" minOccurs="0" maxOccurs="unbounded" />
+    </xs:sequence>
+  </xs:complexType>
+  <xs:simpleType name="LibraryScope">
+    <xs:restriction base="xs:string">
+      <xs:enumeration value="" />
+      <xs:enumeration value="global" />
+      <xs:enumeration value="test suite" />
+      <xs:enumeration value="test case" />
+    </xs:restriction>
+  </xs:simpleType>
+  <xs:simpleType name="DocumentationFormat">
+    <xs:restriction base="xs:string">
+      <xs:enumeration value="ROBOT" />
+      <xs:enumeration value="HTML" />
+      <xs:enumeration value="TEXT" />
+      <xs:enumeration value="REST" />
+    </xs:restriction>
+  </xs:simpleType>
+  <xs:simpleType name="LibraryType">
+    <xs:restriction base="xs:string">
+      <xs:enumeration value="library" />
+      <xs:enumeration value="resource" />
+    </xs:restriction>
+  </xs:simpleType>
+  <xs:simpleType name="GenerationDate">
+    <xs:restriction base="xs:string">
+      <xs:pattern value="\d{8} ([0-1]\d|2[0-3]):[0-5]\d:[0-5]\d" />
```

Comment:
IMHO this is overly complicated and just leaves readers scratch their head. `\d{8} \d\d:\d\d:\d\d` would be a lot easier to understand. Notice that this schema is mainly for documentation purposes and thus more for humans than for machines. Do you have any use case where the exact time validation would bring benefits?

Reply:
Alternatively, should we change the generation time to a date time that is valid `xsd:dateTime`? I seriously doubt anyone uses the time for anything at the moment, so changing it ought to be pretty well backwards compatible. I wouldn't change it in a minor release but in this major release with other spec changes that ought to be fine.

Reply:
Changed the date format in #3528. This part of the spec should now be updated accordingly. I f I understand this correctly, the whole `GenerationDate` element can be removed and `xsd:dateTime` be used in its place.

This change obviously doesn't affect the other schema. In there I'd still simplify the validation so that it's just `\d{8} \d\d:\d\d:\d\d`.

Reply:
i like this change!
done!